### PR TITLE
Code Coverage to %50

### PR DESCRIPTION
**Pull request by metinavci**

### CODE REVIEW

File: SPIClient.Test/PurchaseTest.cs [hunk 1]
```diff
@@ -49,637 +55,730 @@ public void TestPurchaseRequestWithFull()
...
+            Assert.Equal("............5581", response.GetMaskedPan());
+            Assert.Equal("100612348842", response.GetTerminalId());
             Assert.False(response.WasCustomerReceiptPrinted());
             Assert.False(response.WasMerchantReceiptPrinted());
-            Assert.Equal(response.GetSettlementDate(), DateTime.ParseExact(msg.GetDataStringValue("bank_settlement_date"), "ddMMyyyy", CultureInfo.InvariantCulture).Date);
+            Assert.Equal(DateTime.ParseExact(msg.GetDataStringValue("bank_settlement_date"), "ddMMyyyy", CultureInfo.InvariantCulture).Date, response.GetSettlementDate());
             Assert.Equal(response.GetResponseValue("pos_ref_id"), response.PosRefId);
 
+            // act
             response = new RefundResponse();
```
Comment (RaymondK-AP) [inline SPIClient.Test/PurchaseTest.cs]: Feels like this needs to be in another unit test - leave it for now

File: SPIClient.Test/SpiModelsTest.cs [hunk 2]
```diff
@@ -6,192 +6,232 @@ namespace Test
...
+            // arrange
             var stlEnqMsg = new SettlementEnquiryRequest(RequestIdHelper.Id("stlenq")).ToMessage();
             var transactionFlowState = new TransactionFlowState("1", TransactionType.SettlementEnquiry, 0, stlEnqMsg, $"Waiting for EFTPOS connection to make a settlement enquiry");
+
+            // act
             transactionFlowState.CancelFailed("CancelFailed");
 
+            // assert
             Assert.False(transactionFlowState.AttemptingToCancel);
             Assert.Equal(transactionFlowState.DisplayMessage, "CancelFailed");
```
Comment (RaymondK-AP) [inline SPIClient.Test/SpiModelsTest.cs]: The parameters needs to be swapped around

File: SPIClient.Test/ComWrapperTest.cs [hunk 3]
```diff
@@ -11,439 +11,467 @@ namespace Test
...
-            List<OpenTablesEntry> openTablesEntries = (List<OpenTablesEntry>)SpiClientTestUtils.GetInstanceField(typeof(GetOpenTablesCom), getOpenTablesCom, "OpenTablesList");
+            var openTablesEntries = (List<OpenTablesEntry>)SpiClientTestUtils.GetInstanceField(typeof(GetOpenTablesCom), getOpenTablesCom, "OpenTablesList");
 
+            // assert
             Assert.Equal(openTablesEntries[0].TableId, openTablesEntry.TableId);
             Assert.Equal(openTablesEntries[0].Label, openTablesEntry.Label);
             Assert.Equal(openTablesEntries[0].BillOutstandingAmount, openTablesEntry.BillOutstandingAmount);
 
+            // act
             string jsonStr = getOpenTablesCom.ToOpenTablesJson();
```
[the remark this answers is not in the document]
Comment (metinavci) [reply SPIClient.Test/ComWrapperTest.cs]: if we clarified the standards of Unit Test then we could add correct and healthy Unit Tests.

They are separated now.

File: SPIClient.Test/ComWrapperTest.cs [hunk 4]
```diff
@@ -11,439 +11,467 @@ namespace Test
...
+            Assert.Equal(1000, response.GetBankNonCashAmount());
+            Assert.Equal("Visa", response.GetSchemeName());
+            Assert.Equal("Visa", response.GetSchemeApp());
+            Assert.Equal(0, response.GetAmount());
+            Assert.Equal(0, response.GetTransactionAmount());
+            Assert.Equal("14062019153747", response.GetBankDateTimeString());
+            Assert.Equal("190614001137", response.GetRRN());
+            Assert.Equal("APPROVED", response.GetResponseText());
+            Assert.Equal("000", response.GetResponseCode());
+            Assert.Equal(msg.GetDataStringValue("merchant_receipt"), msg.GetDataStringValue("customer_receipt"));
```
[the remark this answers is not in the document]
Comment (metinavci) [reply SPIClient.Test/ComWrapperTest.cs]: They are separated now.

**Pull request closed without merging**